masonry/src/passes/update.rs

```diff
@@ -145,3 +146,51 @@ pub(crate) fn run_update_pointer_pass(root: &mut RenderRoot, root_state: &mut Wi
     // Pass root widget state to synthetic state create at beginning of pass
     root_state.merge_up(root.widget_arena.get_state_mut(root.root.id()).item);
 }
+
+// ----------------
+
+fn update_anim_for_widget(
+    global_state: &mut RenderRootState,
+    mut widget: ArenaMut<'_, Box<dyn Widget>>,
+    mut state: ArenaMut<'_, WidgetState>,
+    elapsed_ns: u64,
+) {
+    let _span = widget.item.make_trace_span().entered();
+
+    if !state.item.needs_anim {
+        return;
+    }
+    state.item.needs_anim = false;
+
+    if state.item.request_anim {
+        state.item.request_anim = false;
```

Comment:
Maybe we should have a comment here that most passes reset their `needs` and `request` after the call to the widget method, but that it's valid and expected for `request_anim` to be called in response to `AnimFrame`.

Reply:
Done.